nemo_aligner/models/nlp/gpt/gpt_reward_model.py

```diff
@@ -235,5 +235,8 @@ def sharded_state_dict(self, prefix=""):
 
         if not self.return_rm_head_in_state_dict:
             sharded_state_dict = {k: v for k, v in sharded_state_dict.items() if "rm_head" not in k}
+        else:
+            assert self.config.fp8 is None, "fp8 is not supported for the reward model"
+            sharded_state_dict = {k: v for k, v in sharded_state_dict.items() if "rm_head._extra_state" not in k}
```

Comment:
Can you add a comment to explain why we need to do this?

Reply:
e17f69a5f11e60a58f10994e2ac173bbd4e39d64